tools/seed_submissions.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 the Deno authors. All rights reserved. MIT license.
+
+import { createItem } from "@/utils/db.ts";
+
+// Reference: https://github.com/HackerNews/API
+const API_BASE_URL = `https://hacker-news.firebaseio.com/v0`;
+
+interface Story {
+  id: number;
+  score: number;
+  time: number;
+  by: string;
+  title: string;
+  url: string;
+}
+
+function* batchify<T>(arr: T[], n = 5): Generator<T[], void> {
+  for (let i = 0; i < arr.length; i += n) {
+    yield arr.slice(i, i + n);
+  }
+}
+
+// Fetch the top 500 HN stories to seed the db
+async function fetchTopStoryIds() {
+  const resp = await fetch(`${API_BASE_URL}/topstories.json`);
+  if (!resp.ok) {
+    console.error(`Failed to fetchTopStoryIds - status: ${resp.status}`);
+    return;
+  }
+  return await resp.json();
+}
+
+async function fetchStory(id: number | string) {
+  const resp = await fetch(`${API_BASE_URL}/item/${id}.json`);
+  if (!resp.ok) {
+    console.error(`Failed to fetchStory (${id}) - status: ${resp.status}`);
+    return;
+  }
+  return await resp.json();
+}
+
+async function fetchTopStories(limit = 10) {
+  const ids = await fetchTopStoryIds();
+  if (!(ids && ids.length)) {
+    console.error(`No ids to fetch!`);
+    return;
+  }
+  const filtered: [number] = ids.slice(0, limit);
+  const stories: Story[] = [];
+  for (const batch of batchify(filtered)) {
+    stories.push(...(await Promise.all(batch.map((id) => fetchStory(id))))
+      .filter((v) => Boolean(v)) as Story[]);
+  }
+  return stories;
+}
+
+async function seedSubmissions(stories: Story[]) {
+  const items = stories.map(({ by: userId, title, url }) => {
+    return { userId, title, url };
+  }).filter(({ url }) => {
+    try {
+      return Boolean(new URL(url).host);
+    } catch {
+      return;
+    }
+  });
```

Comment:
```suggestion
  }).filter(({ url }) => url);
```
Maybe just include posts with a `url` property.